// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.sharkdetector.sentimentanalytic";
option java_outer_classname = "SentimentAnalyticProto";
option objc_class_prefix = "HLW";
option go_package="github.com/chiupc/sentiment_analytic";

package SentimentAnalytic;

service SentimentAnalytic {
    rpc AnalyzeSentiment(InputFile) returns (OutputFile) {}
}

message InputFile{
    string columnName = 1;
    string fileName = 2;
    bytes text = 3;
	string analyzerEngine = 4;
}

message OutputFile{
    string filename = 1;
	string summaryData = 2;
}